src/ProducerInterface.php

```diff
@@ -14,17 +14,21 @@
 
 namespace Graze\Queue;
 
+use Graze\Queue\Adapter\Exception\UnsupportedOperationException;
 use Graze\Queue\Message\MessageInterface;
 
 interface ProducerInterface
 {
     /**
-     * @return MessageInterface
+     * @param string $body
+     * @param array  $options
      */
     public function create($body, array $options = []);
 
     /**
-     * @param MessageInterface[] $message
+     * @param array $message
      */
     public function send(array $messages);
+
+    public function purge();
```

Comment:
Here are my thoughts:

- `Client` implements `ProducerInterface` and `ClientInterface`, so adding `purge()` to `ProducerInterface` does not affect that class negatively.

- You've already got segregated interfaces in `ProducerInterface` and `ClientInterface` and since `Client` implements both, the only consideration really is how people might extend this code. Are we burdening people who might write separate `ProducerInterface` implementations with a method they can't implement? My thinking here is if we could reasonably assume that **all** queue implementations have a way of purging then it's fine to put this on `ProducerInterface`, as a purge operation is similar to a send operation. If we think this assumption is unreasonable then you can segregate further with a `PurgeableInterface`.

What's your thinking on this?